### PR TITLE
Feature/light and vob misc

### PR DESCRIPTION
**Pull request by JucanAndreiDaniel**
can be merged after https://github.com/GothicKit/phoenix-shared-interface/pull/42

### CODE REVIEW

File: PxCs.Tests/PxWorldTest.cs [hunk 1]
```diff
@@ -52,37 +52,70 @@ public void Test_load_Vobs()
...
+
+            // var touchDamage = vobs[0].childVobs!.First(i => i.type == PxWorld.PxVobType.PxVob_oCTouchDamage);
+            // Assert.IsType<PxVobTouchDamageData>(touchDamage);
+
+            // var triggerUntouch = vobs[0].childVobs!.First(i => i.type == PxWorld.PxVobType.PxVob_zCTriggerUntouch);
+            // Assert.IsType<PxVobTriggerUntouchData>(triggerUntouch);
+
+            // var earthquake = vobs[0].childVobs!.First(i => i.type == PxWorld.PxVobType.PxVob_zCEarthquake);
+            // Assert.IsType<PxVobEarthQuakeData>(earthquake);
+
```
Comment (JaXt0r) [inline PxCs.Tests/PxWorldTest.cs:90]: Is it intended to have these tests commented out? Please activate, remove, or comment why they're disabled.

File: PxCs/Data/Vob/PxLightPresetData.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public float coneAngle;
+        public bool isStatic;
+        public PxLightQuality quality;
+        public string lensflareFx = default!;
+
+        public bool on;
+        public float[]? rangeAnimationScale;
+        public float rangeAnimationFps;
+        public bool rangeAnimationSmooth;
+        public Vector4Byte[]? colorAnimationList;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxLightPresetData.cs:21]:
```suggestion
        public Vector4Byte[] colorAnimationList = default!;
```

File: PxCs/Data/Vob/PxVobAnimateData.cs [hunk 3]
```diff
@@ -5,6 +5,6 @@ namespace PxCs.Data.Vob
     [Serializable]
     public class PxVobAnimateData : PxVobData
     {
-
+        public bool startOn;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobAnimateData.cs:8]: That's it? Nothing more? Wow.!🤣

File: PxCs/Data/Vob/PxVobCodeMasterData.cs [hunk 4]
```diff
@@ -5,6 +5,12 @@ namespace PxCs.Data.Vob
     [Serializable]
     public class PxVobCodeMasterData : PxVobData
     {
+        public string target = default!;
+        public bool ordered;
+        public bool firstFalseIsFailure;
+        public string failureTarget = default!;
+        public bool untriggeredCancels;
+        public string[]? slaves;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobCodeMasterData.cs:13]:
```suggestion
        public string[] slaves = default!;
```

File: PxCs/Data/Vob/PxVobData.cs [hunk 5]
```diff
@@ -18,7 +18,7 @@ public class PxVobData
         public PxVobData[] childVobs = default!;
 
         public Vector3 position;
-        public PxMatrix3x3Data rotation = default!;
+        public PxMatrix3x3Data? rotation;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobData.cs:21]: Other files also worked with this  = default! for arrais. Therefor let's have it the same way, please.

```suggestion
        public PxMatrix3x3Data rotation = default!;
```

File: PxCs/Data/Vob/PxVobLightData.cs [hunk 6]
```diff
@@ -5,6 +5,7 @@ namespace PxCs.Data.Vob
     [Serializable]
     public class PxVobLightData : PxVobData
     {
-
+        //since in c# we don't have multiple inheretance we can use this
+        public PxLightPresetData? lightPreset;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobLightData.cs:9]:
```suggestion
        public PxLightPresetData lightPreset = default!;
```

File: PxCs/Data/Vob/PxVobMobContainerData.cs [hunk 7]
```diff
@@ -6,8 +6,8 @@ namespace PxCs.Data.Vob
     public class PxVobMobContainerData : PxVobMobInterData
     {
         public bool locked;
-        public string key;
-        public string pickString;
-        public string contents;
+        public string? key;
+        public string? pickString;
+        public string? contents;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobMobContainerData.cs:11]:
```suggestion
        public string key = default!;
        public string pickString = default!;
        public string contents = default!;
```

File: PxCs/Data/Vob/PxVobNpcData.cs [hunk 8]
```diff
@@ -1,10 +1,81 @@
...
+            public bool used;
+            public string name = default!;
+            public int itemIndex;
+            public bool inInventory;
+        }
+
+        public string npcInstance = default!;
+        public Vector3 modelScale;
+        public float modelFatness;
+        public string[]? overlays;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobNpcData.cs:27]:
```suggestion
        public string[] overlays = default!;
```

File: PxCs/Data/Vob/PxVobNpcData.cs [hunk 9]
```diff
@@ -1,10 +1,81 @@
...
+        public int routineOverlayCount;
+        public int walkmodeRoutine;
+        public bool weaponmodeRoutine;
+        public bool startNewRoutine;
+        public int aiStateDriven;
+        public Vector3 aiStatePos;
+        public string currentRoutine = default!;
+        public bool respawn;
+        public int respawnTime;
+        public int[]? protection;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobNpcData.cs:76]:
```suggestion
        public int[] protection = default!;
```

File: PxCs/Data/Vob/PxVobNpcData.cs [hunk 10]
```diff
@@ -1,10 +1,81 @@
...
+        public string startAiState = default!;
+        public int[]? aivar;
+        public string scriptWaypoint = default!;
+        public int attitude;
+        public int attitudeTemp;
+        public int nameNr;
+        public bool moveLock;
+        public string[]? packed;
+        public PxVobItemData[]? items;
+        public PxVobNpcSlotData[]? slots;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobNpcData.cs:54]:
```suggestion
        public string[] packed = default!;
        public PxVobItemData[] items = default!;
        public PxVobNpcSlotData[] slots = default!;
```

File: PxCs/Data/Vob/PxVobNpcData.cs [hunk 11]
```diff
@@ -1,10 +1,81 @@
...
+        public PxVobNpcTalentData[]? talents;
+        public int fightTactic;
+        public int fightMode;
+        public bool wounded;
+        public bool mad;
+        public int madTime;
+        public bool player;
+        public int[]? attributes;
+        public int[]? hcs;
+        public int[]? missions;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobNpcData.cs:44]:
```suggestion
        public int[] attributes = default!;
        public int[] hcs = default!;
        public int[] missions = default!;
```

File: PxCs/Data/Vob/PxVobNpcData.cs [hunk 12]
```diff
@@ -1,10 +1,81 @@
...
+        public float modelFatness;
+        public string[]? overlays;
+        public int flags;
+        public int guild;
+        public int guildTrue;
+        public int level;
+        public int xp;
+        public int xpNextLevel;
+        public int lp;
+        public PxVobNpcTalentData[]? talents;
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobNpcData.cs:35]:
```suggestion
        public PxVobNpcTalentData[] talents = default!;
```

File: PxCs/Data/Vob/PxVobNpcData.cs [hunk 13]
```diff
@@ -1,10 +1,81 @@
 ﻿using System;
+using System.Numerics;
 
 namespace PxCs.Data.Vob
 {
     [Serializable]
     public class PxVobNpcData : PxVobData
     {
+        public class PxVobNpcTalentData
```
Comment (JaXt0r) [inline PxCs/Data/Vob/PxVobNpcData.cs:9]: Why are these inner classes?
Is it for readability? (As we normally don't use inner classes)

**Review by JaXt0r — CHANGES_REQUESTED** [review]
Tests were all green!
Mostly  = default! change requests.

File: PxCs/Data/Vob/PxVobNpcData.cs (hunk 13 above)
Comment (JucanAndreiDaniel) [reply PxCs/Data/Vob/PxVobNpcData.cs:9]: These inner classes are used only inside this class, and yeah, for readability

**Review by JaXt0r — CHANGES_REQUESTED** [review]
A few naming conventions and PxVmNpc already exists (with less properties). Please check comments and thank you for the additional properties.

File: PxCs/Data/Mesh/PxLightMapData.cs [hunk 14]
```diff
@@ -0,0 +1,12 @@
+using System.Numerics;
+using PxCs.Data.Texture;
+
+namespace PxCs.Data.Mesh
+{
+    public class PxLightMapData
```
[the remark this answers is not in the document]
Comment (JucanAndreiDaniel) [reply PxCs/Data/Mesh/PxLightMapData.cs:6]: This is in mesh namespace as it is used by meshes

File: PxCs/Data/Vob/PxLightPresetData.cs [hunk 15]
```diff
@@ -0,0 +1,26 @@
+﻿using PxCs.Data.Struct;
+using static PxCs.Interface.PxWorld;
+
+namespace PxCs.Data.Vob
+{
+    public class PxLightPresetData
```
[the remark this answers is not in the document]
Comment (JucanAndreiDaniel) [reply PxCs/Data/Vob/PxLightPresetData.cs:6]: this is both a vob and a light, inside phoenix it is declared as such
```
struct Light : public VirtualObject, public LightPreset 
```

File: PxCs/Data/Mesh/PxLightMapData.cs (hunk 14 above)
[the remark this answers is not in the document]
Comment (JucanAndreiDaniel) [reply PxCs/Data/Mesh/PxLightMapData.cs:6]: I believe you meant to comment to PxLightPresetData:
LightPreset is also defined under VOb in phoenix (now zenkit)

File: PxCs/Data/Vob/PxVobNpcData.cs [hunk 16]
```diff
@@ -1,10 +1,81 @@
 ﻿using System;
+using System.Numerics;
 
 namespace PxCs.Data.Vob
 {
     [Serializable]
     public class PxVobNpcData : PxVobData
```
[the remark this answers is not in the document]
Comment (JucanAndreiDaniel) [reply PxCs/Data/Vob/PxVobNpcData.cs:7]: Duh, you're totally right, my mistake

[the remark this answers is not in the document]
Comment (JucanAndreiDaniel) [reply PxCs/Data/Vob/PxVobNpcData.cs:7]: or not, check please https://github.com/GothicKit/ZenKit/blob/main/include/zenkit/vobs/Misc.hh#L229C3-L229C3

File: PxCs/Data/Vob/PxLightPresetData.cs (hunk 15 above)
[the remark this answers is not in the document]
Comment (lmichaelis) [reply PxCs/Data/Vob/PxLightPresetData.cs:6]: Light presets are not VObs but the have the same content as lights. Notice that `LightPreset` does not inherit from `VirtualObject`

**Comment by JaXt0r** [comment]
I'm currently switching to ZenKit's C# implementation. You can keep this for now, but you might consider checking it, if your implementation is there already.

**Comment by lmichaelis** [comment]
This PR is now obsolete due to the switch to [ZenKitCS](https://github.com/GothicKit/ZenKitCS).

**Pull request closed without merging**